tests/test_experimental.py

```diff
@@ -431,6 +431,92 @@ def test_that_cov_YY_can_be_computed_outside_of_assimilate(
 
             X_i = X_i1
 
+    @pytest.fixture()
+    def large_linear_problem(self):
+        """
+        Creates a problem large enough to realistically benchmark parallelization.
+        """
+        rng = np.random.default_rng(42)  # Use a fixed seed for reproducibility
+
+        num_parameters = 5000
+        num_observations = 200
+        num_ensemble = 200
+
+        A = np.exp(rng.standard_normal(size=(num_observations, num_parameters)))
+
+        def g(X):
+            """Forward model."""
+            return A @ X
+
+        x_true = np.linspace(-1, 1, num=num_parameters)
+        observations = g(x_true) + rng.standard_normal(size=num_observations) / 10
+
+        X = rng.normal(size=(num_parameters, num_ensemble))
+        covariance = rng.triangular(0.1, 1, 1, size=num_observations)
+        yield X, g, observations, covariance
+
+    def test_parallelization_runtime_comparison(self, large_linear_problem):
+        """
+        Compares the runtime of the assimilate method with and without parallelization.
+        """
+        X, g, observations, covariance = large_linear_problem
+
+        # Use a single assimilation step for a clear comparison
+        alpha = 1.0
+
+        # --- Setup the smoother and common data ---
+        smoother = AdaptiveESMDA(
+            covariance=covariance, observations=observations, seed=1
+        )
+        Y = g(X)
+        D = smoother.perturb_observations(ensemble_size=Y.shape[1], alpha=alpha)
+
+        # Make separate copies for each run to ensure a fair start
+        X_serial = np.copy(X)
+        X_parallel = np.copy(X)
+
+        # --- 1. Serial Execution ---
+        print("\n--- Running Serial Benchmark (n_jobs=1) ---")
+        start_serial = time.perf_counter()
+        X_serial = smoother.assimilate(
```

Comment:
Maybe also another test where we mock `def _update_single_parameter(` with `raise ValueError` and see that it raises?